src/QsCompiler/Compiler/RewriteSteps/Monomorphization.cs

```diff
@@ -71,7 +71,7 @@ private ImmutableArray<QsQualifiedName> AugmentedEntryPoints(QsCompilation compi
         {
             // If this compilation is for a library project, there are no defined entry points. Instead,
             // treat every public, non-generic callable as a possible entry point into the library.
-            return this.isLibrary ?
+            return compilation.EntryPoints.Length == 0 ?
```

Comment:
I suppose the presence or lack of entry points is sufficient to identify a library, since a library by definition can't have entry points and an executable must have at least one. If that's the case and we also do intend to run monorphization for libraries even when QIR is not explicitly enabled, then maybe we no longer need `enabledForLibraries` at all?

Reply:
I decided to remove the precondition and the parameter for both the monomorphization and the syntax tree trimming.